src/utilities/activityContent.ts

```diff
@@ -0,0 +1,63 @@
+import path from "path";
+import {
+  createVideo,
+  createImage,
+  ActivityContentNote,
+  ActivityContentAttachment,
+} from "@dsnp/sdk/core/activityContent";
+
+export const noteToActivityContentNote = (
+  note: string,
+  uriList: string[]
+): ActivityContentNote => {
+  const activityContent: ActivityContentNote = {
+    "@context": "https://www.w3.org/ns/activitystreams",
+    type: "Note",
+    content: note,
+    published: new Date().getTime().toString(16),
+  };
+  if (uriList) {
+    if (typeof uriList === "string") {
+      uriList = [uriList];
+    }
+    activityContent["attachment"] = uriList.map(
+      (item): ActivityContentAttachment => {
+        const extension = path.extname(item).replace(".", "");
+        const supportedVideoDomains = /youtu.be|youtube.com|vimeo.com/;
+        if (supportedVideoDomains.test(item)) {
+          return createVideo(item, "text/html", { algorithm: "" }, 0, 0);
```

Comment:
maybe the logic for the type of video could be extracted to its own function

Reply:
good call. extracted.